src/main/java/com/exactpro/th2/sailfish/utils/filter/EqualityFilter.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.sailfish.utils.filter;
+
+import com.exactpro.sf.aml.scriptutil.ExpressionResult;
+import com.exactpro.sf.aml.scriptutil.MvelException;
+import com.exactpro.sf.common.messages.IMessage;
+import com.exactpro.th2.common.grpc.FilterOperation;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+import java.util.Objects;
+import java.util.function.Function;
+
+import static com.exactpro.th2.sailfish.utils.filter.util.FilterUtils.convertToComparableValue;
+import static com.exactpro.th2.sailfish.utils.filter.util.FilterUtils.getObjectType;
+
+
+public class EqualityFilter implements IOperationFilter {
+
+    private FilterOperation operation;
+    private final Comparable<?> value;
+    private Function<Object, Boolean> comparator;
+
+
+    public EqualityFilter(@NotNull String simpleFilter, @NotNull FilterOperation filterOperation) {
+        this.value = Objects.requireNonNull(convertToComparableValue(simpleFilter), "Value cannot be converted or null");
```

Comment:
!! Why do you need `convertToComparableValue` here? We suppose to check the values without converting them

Reply:
I think we should to convert the expected value because we can to identify value type. Otherwise we will always get `String`  as the value type

Reply:
Unfortunately, there is no way to say whether the `4.5` is a String or Number